express-notes-api/server.js

```diff
@@ -44,18 +44,43 @@ app.post('/api/notes', async (req, res) => {
   } else {
     try {
       const data = await readFile(dataPath, 'utf8');
-      const notes = JSON.parse(data).notes;
-      const id = Math.floor(Math.random() * 100) + 1;
-      notes[id] = { id, content };
-      await writeFile(dataPath, JSON.stringify({ notes }));
-      res.status(201).json(notes[id]);
+      const idNotes = JSON.parse(data);
+      const newId = idNotes.nextId.toString();
```

Comment:
You could just assign `idNotes.nextId` directly and avoid the conversion to a string, especially since you convert it back to a number on line 50. (You can use numbers to index into an object.)